appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java

```diff
@@ -37,91 +36,49 @@ public class NotificationInitializer {
     private static final Logger LOGGER = LoggerFactory
             .getLogger(NotificationInitializer.class);
 
-    private static final String NOTIFIED_UIS_SESSION_PARAM = "vaadin-appsec-kit-notified-uis";
-    /**
-     * New UI instance check interval (in ms). Determines how often the session
-     * is checked for new UI instances which might need a notification shown.
-     */
-    private static final int NOTIFICATION_CHECK_INTERVAL = 5000;
-
     /**
      * Notification timeout (in ms). Set to 24 hours to essentially make it
      * persistent until either the appsec link is clicked or the notification is
      * dismissed.
      */
     private static final int NOTIFICATION_DELAY = 24 * 60 * 60 * 1000;
 
-    private static final Map<String, ScheduledFuture<?>> scheduledChecks = new ConcurrentHashMap<>();
+    private final Map<VaadinSession, Registration> scanEventRegistrations = new ConcurrentHashMap<>();
 
-    /**
-     * Initializes the notification service
-     *
-     * @param vaadinService
-     *            current VaadinService instance
-     */
-    static void serviceInit(VaadinService vaadinService) {
-        if (isDebugMode(vaadinService)) {
-            vaadinService.addSessionInitListener(e -> {
-                e.getSession().addUIProvider(new AppSecUIProvider());
-                startNotificationTask(e.getSession());
-            });
+    public void serviceInit(VaadinService service) {
+        if (isDebugMode(service)) {
+            service.addSessionInitListener(this::subscribeSessionToScanEvents);
+            service.addSessionDestroyListener(this::removeSessionRegistration);
             LOGGER.info("NotificationInitListener initialized.");
```

Comment:
In v7 version there is no `NotificationInitListener` class. Could be this changed to something appropriate for v7 version, e.g. `NotificationInitializer initialized.`?